### DISCUSSION TITLE
Welcome to OpenSimRT Discussions!

### DISCUSSION THREAD

**Discussion by mitkof6** [mitkof6/OpenSimRT#17] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Kristofferson777** [comment]
HI, I'm confused on how to get OpensimRT to work. I understand how to work with models in opensim and would like to take positional data and then calculated kinetic data from that positional data. From there I would like to take that kinematic data along with a scaled or premade opensim model and calculate grf. I'm not quite sure how to do that though with this project. I was hoping you could guide me through the process.

**Reply by Kristofferson777** [reply]
Essentially I'm asking how to I go from a marker based video to getting ground reaction force data with this project.

**Reply by Kristofferson777** [reply]
When trying to build "testAccelerationGRFMPredictionFromFIle.cpp" I get the following error:

CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
[cmake] Please set them or make sure they are set and tested correctly in the CMake files:
[cmake] oscpack
[cmake]     linked by target "IMU" in directory C:/Users/Chris/OneDrive/Desktop/OpenSimRT-master/OpenSimRT/IMU
[cmake]     linked by target "TestLowerLimbIMUIKFromFile" in directory C:/Users/Chris/OneDrive/Desktop/OpenSimRT-master/OpenSimRT/IMU
[cmake]     linked by target "TestUpperLimbIMUIKFromFile" in directory C:/Users/Chris/OneDrive/Desktop/OpenSimRT-master/OpenSimRT/IMU
[cmake]     linked by target "OnlineLowerLimbIMUIK" in directory C:/Users/Chris/OneDrive/Desktop/OpenSimRT-master/OpenSimRT/IMU
[cmake]     linked by target "OnlineUpperLimbIMUIK" in directory C:/Users/Chris/OneDrive/Desktop/OpenSimRT-master/OpenSimRT/IMU

**Reply by mitkof6** [reply]
The GRF prediction module is experimental and might not work out of the box for any problem. Predicting GRF from kinematics is an underdetermined problem that has infinitely many solutions. You are encouraged to test our example but I see you have difficulties building the project, which is understandable due to the complexity of the C++ building system.

**Comment by Kristofferson777** [comment]
I'm not a coder and I'm pretty new to opensim custom tools and plugins setup. I have a great need for the ability to perform ground reaction force prediction. I was hoping you could make a more detailed install instruction file or reply with how I am supposed to set up the files and run or build/run them in visual studio or otherwise. It's not clear how to go from downloading the files to running the program . Thank you in advance.

**Reply by Kristofferson777** [reply]
I'm getting a string loo long error in MVS when running the tests projects

**Reply by mitkof6** [reply]
Building the project on Windows might be more challenging if you are not familiar with CMake and building OpenSim dependencies process. There are a lot of steps and details. I will encourage you to try to build the project on Linux using our automated script. If you are able to build the project, then you can test the examples, and possibly adapt them to your needs.

**Reply by mitkof6** [reply]
Here is the steps to build everything that are used by GitHub CI https://github.com/mitkof6/OpenSimRT/blob/master/.github/workflows/continuous-integration.yml

**Reply by mitkof6** [reply]
And here are the environment variables https://github.com/mitkof6/OpenSimRT/blob/master/.github/workflows/env_variables

**Reply by mitkof6** [reply]
Basically, you need to isolate only the commands under run tag. The project was developed and tested with Ubuntu 18.04. If you use another distribution you might encounter errors since dependencies are pre-build by us for this specific distribution. You can of course build all the dependencies manually but this might be challenging.

**Reply by mitkof6** [reply]
Here I have helper automated scripts for building the dependencies manually https://github.com/mitkof6/OpenSimRT/tree/master/dependencies

**Reply by mitkof6** [reply]
Apart from that, I cannot guide you more on how to build the project. We tried to automate the building process on Linux, but still it is challenging if you have not been into this before. Good luck 🤞!

**Reply by Kristofferson777** [reply]
Thank you so much for the explanations and help. I have setup Ubuntu 18.04 on a virtual drive on my computer. I believe I have successfully done all the steps in the build_opensim-core file and the build_opensim.md file. I am now trying to run the TestAccelerationGRFMPrediction.cpp file using the g++ command. I am having trouble with the #includes as the terminal can't find the include files. I tried using the include commands but they only worked for the first three dependencies. How do you run the .cpp's with all their dependencies?

**Comment by Kristofferson777** [comment]
Hey mitkof6 I have some questions for you regarding getting opensimrt to work. We setup Linux 18.04 and successfully downloaded opensimrt and opensimcore. We also successfully ran the build files in the dependencies folder. We installed and/or built the additional dependencies that were required like casadi and tropter. We also successfully compiled and generated files for opensimrt using cmake. We also successfully ran the test command at the end of the continuous integration file via the terminal and got no failures. My question now is how do we actually run opensimrt? First off this is all built on Linux as the windows distribution is not properly configured so we don't have a .exe to run the program. Just running the programs via the terminal doesn't bring up visualizer or generate documents. So how do we recreate the visualizer and results as shown in your demo video. I'm specifically asking how to actually run opensimrt, do we need a visualization engine like unity, do we run it through some command?

**Reply by mitkof6** [reply]
Well done! When you build the OpenSimRT project, there is a flag in CMake which enables or disables visualization. For testing during CI this is disabled. If you enable it and rebuild you will be able to visualize the tests using Somebody's buildin visualizer. The GUI that you saw in the video is not part of this project and unfortunately it was not made public due to lack of time to groom the code and document things.

**Comment by Kristofferson777** [comment]
Hello mitkof6, it's been a while since last I've messaged here. I never did get OpensimRT to work and ended moving on to other large tasks in hopes of finding a ground reaction force prediction software alternative. However, I have yet to find a good solution and have now dead ended my alternative tasks and now am at an awkward point where I really need a working solution but have yet to find one. I was wondering if you yourself, or if you know of someone else that has a working copy of OpensimRT that either can be easily exported and installed or knows/has a detailed installation guide for setting it up. I'm in great need of a software like opensimrt that can predict ground reaction forces and was hoping you might be able to help me.

**Comment by Kristofferson777** [comment]
Hi mitkof6 I've managed to get OpensimRT downloaded and built on a Linux 18.04 VM. However I'm not sure how I am supposed to run the GRF prediction. How do you actually run the GRF prediction through opensimRT with both the test data (gait 2392) and my own data?
